spring-cloud-sleuth-core/src/main/java/org/springframework/cloud/sleuth/sampler/PercentageBasedSampler.java

```diff
@@ -0,0 +1,46 @@
+package org.springframework.cloud.sleuth.sampler;
+
+import lombok.extern.slf4j.Slf4j;
+import org.springframework.cloud.sleuth.Sampler;
+import org.springframework.cloud.sleuth.Span;
+import org.springframework.cloud.sleuth.TraceAccessor;
+
+/**
+ * Sampler that based on the given percentage rate will allow sampling.
+ *
+ * A couple of assumptions have to take place in order for the algorithm to work properly:
+ *
+ * <ul>
+ *     <li>We're taking the TraceID into consideration for sampling to be consistent</li>
+ *     <li>We apply the Zipkin algorithm to define whether we should sample or not (we're doing a mod 100) - https://github.com/openzipkin/zipkin-java/blob/master/zipkin-java-core/src/main/java/io/zipkin/TraceIdSampler.java#L78</li>
```

Comment:
sorry to say.. doc's wrong.. not doing mod anymore.. "we're comparing against a threshold"